File: 1.0/src/cz/mg/vulkan/PFNvkCmdBindPipeline.java
package cz.mg.vulkan;

public class PFNvkCmdBindPipeline extends VkFunctionPointer {
    public PFNvkCmdBindPipeline() {
    }

    protected PFNvkCmdBindPipeline(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkCmdBindPipeline(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCmdBindPipeline(long value) {
        setValue(value);
    }

    public PFNvkCmdBindPipeline(VkInstance instance) {
        super(instance, new VkString("vkCmdBindPipeline"));
    }

    public void call(VkCommandBuffer commandBuffer, VkPipelineBindPoint pipelineBindPoint, VkPipeline pipeline){
        callNative(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), pipelineBindPoint != null ? pipelineBindPoint.getVkAddress() : VkPointer.getNullAddressNative(), pipeline != null ? pipeline.getVkAddress() : VkPointer.getNullAddressNative());
    }

    protected static native void callNative(long vkaddress, long commandBuffer, long pipelineBindPoint, long pipeline);


    public void call(VkCommandBuffer commandBuffer, int pipelineBindPoint, VkPipeline pipeline){
        callSimplifiedNative(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), pipelineBindPoint, pipeline != null ? pipeline.getVkAddress() : VkPointer.getNullAddressNative());
    }

    protected static native void callSimplifiedNative(long vkaddress, long commandBuffer, int pipelineBindPoint, long pipeline);

}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkCmdBindPipeline.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCmdBindPipeline_callNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jlong pipelineBindPoint, jlong pipeline){
    (void)env;
    (void)jc;
    PFN_vkCmdBindPipeline f = (PFN_vkCmdBindPipeline)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        *((VkPipelineBindPoint*)jniLongToPointer(pipelineBindPoint)),
        *((VkPipeline*)jniLongToPointer(pipeline))
    );
}


void Java_cz_mg_vulkan_PFNvkCmdBindPipeline_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jint pipelineBindPoint, jlong pipeline){
    (void)env;
    (void)jc;
    PFN_vkCmdBindPipeline f = (PFN_vkCmdBindPipeline)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        (VkPipelineBindPoint)pipelineBindPoint,
        *((VkPipeline*)jniLongToPointer(pipeline))
    );
}

